### PR TITLE
tests all url types work as expected on builders

### PR DESCRIPTION
**Pull request by gapra-msft**
Resolves https://github.com/Azure/azure-sdk-for-java/issues/6357

### CODE REVIEW

File: sdk/storage/azure-storage-file-datalake/src/test/java/com/azure/storage/file/datalake/UrlTests.groovy [hunk 1]
```diff
@@ -0,0 +1,98 @@
+package com.azure.storage.file.datalake
+
+class UrlTests extends APISpec {
+
+    def "test custom url"() {
```
Comment (rickle-msft) [inline sdk/storage/azure-storage-file-datalake/src/test/java/com/azure/storage/file/datalake/UrlTests.groovy]: Can you give a little more description of what's supposed to be happening here? Is this asserting that custom urls don't get modified for dfs/blob endpoints?

File: sdk/storage/azure-storage-file-datalake/src/test/java/com/azure/storage/file/datalake/UrlTests.groovy [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+            .endpoint(customUrl)
+            .credential(primaryCredential)
+            .buildClient()
+        then:
+        client.getAccountUrl() == client.blobServiceClient.getAccountUrl()
+    }
+
+    def "test short host url"() {
+        when:
+        def shortHostUrl = "https://account.core.windows.net"
```
Comment (rickle-msft) [inline sdk/storage/azure-storage-file-datalake/src/test/java/com/azure/storage/file/datalake/UrlTests.groovy]: Is this really a valid azure url?

File: sdk/storage/azure-storage-file-datalake/src/test/java/com/azure/storage/file/datalake/UrlTests.groovy [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+        then:
+        serviceClient.getAccountUrl() == dfsUrl
+        serviceClient.blobServiceClient.getAccountUrl() == blobUrl
+        fileSystemClient.getFileSystemUrl() == dfsUrl + "/container"
+        fileSystemClient.blobContainerClient.getBlobContainerUrl() == blobUrl + "/container"
+        pathClient.getPathUrl() == dfsUrl + "/container/blob"
+        pathClient.blockBlobClient.getBlobUrl() == blobUrl + "/container/blob"
+    }
+
+    def "test dfs service url"() {
```
Comment (rickle-msft) [inline sdk/storage/azure-storage-file-datalake/src/test/java/com/azure/storage/file/datalake/UrlTests.groovy]: I think you could parameterize this test to merge it with the previous one

File: sdk/storage/azure-storage-file-datalake/src/test/java/com/azure/storage/file/datalake/UrlTests.groovy (hunk 2 above)
Comment (gapra-msft) [reply sdk/storage/azure-storage-file-datalake/src/test/java/com/azure/storage/file/datalake/UrlTests.groovy]: not a valid Azure URL, just testing an edge case

File: sdk/storage/azure-storage-file-datalake/src/test/java/com/azure/storage/file/datalake/UrlTests.groovy (hunk 1 above)
Comment (gapra-msft) [reply sdk/storage/azure-storage-file-datalake/src/test/java/com/azure/storage/file/datalake/UrlTests.groovy]: yeah, essentially anything that isnt a .blob. or .dfs. type of URL should not get modified by the builders

File: sdk/storage/azure-storage-file-datalake/src/test/java/com/azure/storage/file/datalake/UrlTests.groovy [hunk 4]
```diff
@@ -0,0 +1,98 @@
+package com.azure.storage.file.datalake
+
+class UrlTests extends APISpec {
+
+    def "test custom url"() {
+        when:
+        def customUrl = "https://www.customstorageurl.com"
```
Comment (alzimmermsft) [inline sdk/storage/azure-storage-file-datalake/src/test/java/com/azure/storage/file/datalake/UrlTests.groovy]: What is the expected return for `accountName` in this case?

Comment (gapra-msft) [reply sdk/storage/azure-storage-file-datalake/src/test/java/com/azure/storage/file/datalake/UrlTests.groovy]: Custom URLs dont actually work for Datalake, so the user should get some exception probably a 404

File: sdk/storage/azure-storage-file-datalake/src/test/java/com/azure/storage/file/datalake/UrlTests.groovy (hunk 1 above)
Comment (gapra-msft) [reply sdk/storage/azure-storage-file-datalake/src/test/java/com/azure/storage/file/datalake/UrlTests.groovy]: yes, I'll add some comments to each test

File: sdk/storage/azure-storage-file-datalake/src/test/java/com/azure/storage/file/datalake/UrlTests.groovy (hunk 3 above)
Comment (gapra-msft) [reply sdk/storage/azure-storage-file-datalake/src/test/java/com/azure/storage/file/datalake/UrlTests.groovy]: just parameterized all the tests using unroll

**Pull request merged**